tasks/main.yml

```diff
@@ -121,15 +125,11 @@
     - '{{ ferm_input_dependent_list }}'
   when: (ferm_enabled|bool and (item.type is defined and item.type) and
          (item.delete is undefined or not item.delete | bool))
+  notify: [ 'Restart ferm' ]
   tags: [ 'role::ferm:rules' ]
 
-- name: Apply iptables rules if ferm is enabled
-  service:
-    name: 'ferm'
-    state: 'restarted'
-  changed_when: False
-  when: ferm_enabled | bool
-  tags: [ 'role::ferm:rules' ]
+- name: Flush handlers to restart ferm if changes were made
+  meta: flush_handlers
```

Comment:
You need also to add the condition and tag, so that when users run this tag, flush handlers is included:
```
- name: Flush handlers to restart ferm if changes were made
  meta: flush_handlers
  when: ferm_enabled | bool
  tags: [ 'role::ferm:rules' ]
```